### PR TITLE
DOC: Fix EX01 issues in docstrings

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Towards #37875

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -5723,6 +5735,17 @@ def to_period(self, freq: str | None = None, copy: bool | None = None) -> Series
         -------
         Series
             Series with index converted to PeriodIndex.
+
+        Examples
+        --------
+        >>> idx = pd.DatetimeIndex(['2015-01-01', '2015-01-02', '2015-01-03'])
+        >>> idx = idx.to_period()
```
Comment (mroeschke) [inline pandas/core/series.py:5742]: This should show `to_period` being called on `Series` e.g.

```
ser = pd.Series(range(3), index=pd.date_range("2023", periods=3))
ser.index
ser.to_period().index
```

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me pending green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @DeaMariaLeon (test failures unrelated)

**Comment by DeaMariaLeon** [comment]
Thank you.

**Pull request merged**